### PR TITLE
Fixes #1542 add operation mode

### LINKED ISSUE #1542: Add "Operation" mode to the DescriptorCriteria
Default AND, can be switched to OR
That way, we do not have to care about grammar

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: src/OSPSuite.Core/Domain/Descriptors/DescriptorCriteria.cs [hunk 1]
```diff
@@ -5,16 +5,33 @@
...
+   public enum DescriptorCriteriaOperator
+   {
+      And,
+      Or
+   }
+
    public class DescriptorCriteria : List<IDescriptorCondition>, ISpecification<IEntity>, ISpecification<EntityDescriptor>
    {
+      //Default behavior
+      public DescriptorCriteriaOperator Operator { get; set; } = DescriptorCriteriaOperator.And;
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Descriptors/DescriptorCriteria.cs:17]: @Yuri05 As discussed

File: src/OSPSuite.Core/Domain/Descriptors/DescriptorCriteria.cs [hunk 2]
```diff
@@ -5,16 +5,33 @@
...
 
       public bool IsSatisfiedBy(EntityDescriptor entityDescriptor)
       {
-         return this.Any() && this.All(condition => condition.IsSatisfiedBy(entityDescriptor));
+         //Empty criteria does not fulfill 
+         if (!this.Any())
+            return false;
+
+         if (Operator == DescriptorCriteriaOperator.And)
+            return this.All(x => x.IsSatisfiedBy(entityDescriptor));
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Descriptors/DescriptorCriteria.cs:32]: what we had before

File: src/OSPSuite.Core/Domain/Descriptors/DescriptorCriteria.cs [hunk 3]
```diff
@@ -5,16 +5,33 @@
...
+
+         return this.Any(x => x.IsSatisfiedBy(entityDescriptor));
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Descriptors/DescriptorCriteria.cs:34]: the OR

**Comment by msevestre** [comment]
Merging @Yuri05 to work on MoBi

File: src/OSPSuite.Core/Domain/Constants.cs [hunk 4]
```diff
@@ -568,6 +568,7 @@ public static class ParameterExport
 
       public const string NOT = "Not";
       public const string AND = "and";
+      public const string OR = "or";
```
Comment (Yuri05) [inline src/OSPSuite.Core/Domain/Constants.cs:571]: why are "or" and "and" written in small letters and "Not" not? 😄

**Review by Yuri05 — COMMENTED** [review]
looks good :)

File: src/OSPSuite.Core/Domain/Constants.cs (hunk 4 above)
Comment (msevestre) [reply src/OSPSuite.Core/Domain/Constants.cs:571]: good point. At the end of the day, we show them in ALL CAPS. In can update next time

**Pull request merged**